lib/rust/flexer/src/lib.rs

```diff
@@ -14,26 +14,241 @@
 //! These lexers are capable of lexing any regular grammar, with some extensions to allow working
 //! with context sensitive (e.g. indentation-aware) syntax.
 
-pub use enso_prelude as prelude;
+// TODO [AA] Logging https://github.com/enso-org/ide/blob/main/src/rust/ide/src/model/execution_context/synchronized.rs#L45
 
+use crate::prelude::*;
 
+use lazy_reader::LazyReader;
+use crate::group::GroupRegistry;
 
 pub mod automata;
-pub mod group;
 pub mod data;
+pub mod group;
+
+#[allow(missing_docs)]
+pub mod prelude {
+    pub use enso_prelude::*;
+}
 
 
 
-// ===============
-// === Flexer ====
-// ===============
+// =================
+// === Constants ===
+// =================
+
+mod constants {
+    /// The number of 'frames' to reserve in the state stack.
+    pub const STATE_STACK_RESERVATION:usize = 1024;
+    /// The size of the output buffer (in tokens) to reserve.
```

Comment:
as above - needed explanation why this number is important and what happens if its too low.